From 11d4f925f4b5c0a6d6ed97c0a09c66fb1f11342f Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Thu, 30 Mar 2017 16:44:36 +0300
Subject: [PATCH 1540/2241] fix: dts: ap806: Fix broken AP SDHCI interface
 functionality

Add back the Ap SDHCI interface limitations to AP-806 DTS
	marvell,xenon-phy-slow-mode;
	max-frequency = <50000000>;
Without this change the SD/eMMC device detection om AP side
fails with Xenon v6 driver.
Currently the driver distinguish between AP and CP eMMC
interfaces with a help of compatibility string, however
without asking Xenon PHY to work in a slow mode, the card
detection fails.
Need more investigation for closing this issue inside the
driver code without need for these DTS entries.

Change-Id: I5538f4a977c1b6f5c2ac136144e2e822d1315b26
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/38231
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: Omri Itach <omrii@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-ap806.dtsi | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
index 3372c0a..a41f56f 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
@@ -350,6 +350,8 @@
 				 * The AP SDIO interface should work in PHY slow mode
 				 * and clock not higher than 50MHz due to HW limitation
 				 */
+				marvell,xenon-phy-slow-mode;
+				max-frequency = <50000000>;
 				dma-coherent;
 				status = "disabled";
 			};
-- 
2.7.4

